flaky test case in windows:  ThreadSanitizer failed to allocate 0x000003f29000 (66228224) bytes at 0x200dbdaa74000 (error code: 87)

### Description

https://github.com/containerd/containerd/runs/7052774182?check_suite_focus=true

```
=== Failed
=== FAIL: pkg/cri/server  (0.00s)
==5044==ERROR: ThreadSanitizer failed to allocate 0x000003f29000 (66228224) bytes at 0x200dbdaa74000 (error code: 87)
FAIL	github.com/containerd/containerd/pkg/cri/server	0.120s
DONE 612 tests, 9 skipped, 1 failure in 178.444s
mingw32-make: *** [makefile:206: test] Error 1
```

No sure what happen in pkg/cri/server in windows. The CI is unstable right now.

### Steps to reproduce the issue

No response

### Describe the results you received and expected

No response

### What version of containerd are you using?

No response

### Any other relevant information

_No response_

### Show configuration if it is related to CRI plugin.

_No response_

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -281,6 +281,14 @@ jobs:
 
       - run: script/setup/install-dev-tools
 
+      # There is currently an issue in the race detector in Go on Windows when
+      # used with a newer version of GCC. The issue was first reported here:
+      # https://github.com/golang/go/issues/46099
+      - name: Downgrade MinGW
+        shell: bash
+        run: |
+          choco install mingw --version 10.2.0 --allow-downgrade
+
       - name: Binaries
         env:
           CGO_ENABLED: 1

